core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -166,7 +179,32 @@ class DefaultAutoTopicCreationManager(
         debug(s"Auto topic creation completed for ${creatableTopics.keys}.")
         clearInflightRequests(creatableTopics)
       }
-    })
+    }
+
+    val channelManager = this.channelManager.getOrElse {
+      throw new IllegalStateException("Channel manager must be defined in order to send CreateTopic requests.")
+    }
+
+    val request = metadataRequestContext.map { context =>
+      val requestVersion =
+        channelManager.controllerApiVersions() match {
+          case None =>
+            ApiKeys.CREATE_TOPICS.latestVersion()
```

Comment:
Let's add a comment that we rely on the Metadata request to be retried in the case that the latest version is not usable by the controller.